src/pages/storage/forms/StorageVolumeForm.tsx

```diff
@@ -79,6 +83,10 @@ export const volumeFormToPayload = (
       [getVolumeKey("zfs_remove_snapshots")]: values.zfs_remove_snapshots,
       [getVolumeKey("zfs_use_refquota")]: values.zfs_use_refquota,
       [getVolumeKey("zfs_reserve_space")]: values.zfs_reserve_space,
+      ...getUnhandledKeyValues(
+        (volume?.config as LxdConfigPair) || {},
+        getVolumeConfigKeys(),
+      ),
     },
     project,
```

Comment:
We are handling unknown config keys, but there might also be main level keys missing. I.e. see in `src/pages/profiles/EditProfile.tsx:146` how we enrich config and main level keys from the original entity.

Also, might be easier to read, if we evaluate `getVolumeConfigKeys` above the return block and use a const in it.